src/services/expense.service.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+let expenses = [];
+let expId = 1;
+
+const generateExpId = () => {
+  const newId = expId;
+
+  expId++;
+
+  return newId;
+};
+
+const getAll = () => {
+  return expenses;
+};
+
+const getFiltered = (userId, categories, fromDate, toDate) => {
+  let filterData = [...expenses];
+
+  if (userId) {
+    filterData = filterData.filter(ex => ex.userId === Number(userId));
+  }
+
+  if (categories) {
+    filterData = filterData.filter(ex => categories.includes(ex.category));
+  }
+
+  if (fromDate) {
+    const dateFrom = new Date(fromDate);
+
+    filterData = filterData.filter(ex => new Date(ex.spentAt) >= dateFrom);
+  }
+
+  if (toDate) {
+    const dateTo = new Date(toDate);
+
+    filterData = filterData.filter(ex => new Date(ex.spentAt) <= dateTo);
+  }
```

Comment:
please use condition in one `.filter` method, because now it will invoke filter a lot of times of there is a multiple filters in query